### PR TITLE
Kernel: Implement DiskPartition to prepare for partition support

### PR DESCRIPTION
**Pull request by deoxxa**
This implements a passthrough disk driver that translates the read/write
block addresses by a fixed offset. This could form the basis of MBR
partition support if we were to parse the MBR table at boot and create that
OffsetDiskDevice dynamically, rather than seeking to a fixed offset.

This also introduces a dependency in the form of grub. You'll need to have
32-bit grub binaries installed to build the project now.

As a bonus, divorcing Serenity from qemu's kernel loading means we can now
*technically* boot on real hardware. It just... doesn't get very far yet.
If you write the `_disk_image` file to an IDE hard drive and boot it in a
machine that supports all the basic PC hardware, it *will* start loading
the kernel.

### CODE REVIEW

File: Kernel/sync.sh [hunk 1]
```diff
@@ -1,33 +1,68 @@
...
 if [ $(id -u) != 0 ]; then
-    echo "This needs to be run as root"
-    exit 1
+	die "this script needs to run as root"
+fi
+
+echo "setting up disk image..."
+if [ ! -f _disk_image ]; then
+	echo "not found; creating a new one"
+  dd if=/dev/zero of=_disk_image bs=1M count=100 || die "couldn't create disk image"
```
Comment (awesomekling) [inline Kernel/sync.sh]: 100MB is too small if you start building stuff in the `Ports` directory. We're gonna need a bigger default, but maybe there could be a command-line option for overriding the disk image size?

File: Kernel/Devices/OffsetDiskDevice.h [hunk 2]
```diff
@@ -0,0 +1,24 @@
+#pragma once
+
+#include <AK/RetainPtr.h>
+#include <Kernel/Devices/DiskDevice.h>
+
+class OffsetDiskDevice final : public DiskDevice {
+public:
+    static Retained<OffsetDiskDevice> create(Retained<DiskDevice>&& device, unsigned offset);
```
Comment (awesomekling) [inline Kernel/Devices/OffsetDiskDevice.h]: Longer-term I think this should be a `DiskPartition` class that has an offset but also a `length`.
We could then have `DiskDevice` read the partition table and instantiate a number of `DiskPartition` objects.

Also maybe we should call the `offset` argument here `block_offset` just to make things clear.

File: Kernel/sync.sh (hunk 1 above)
Comment (deoxxa) [reply Kernel/sync.sh]: Ah yep, I think it was 500MB before. I'll put it back to that, and add an option to change it as suggested.

File: Kernel/Devices/OffsetDiskDevice.h (hunk 2 above)
Comment (deoxxa) [reply Kernel/Devices/OffsetDiskDevice.h]: I did wonder about the naming for this thing. I followed the example from `FileBackedDiskDevice`, but I'm not picky about what it's called.

Comment (awesomekling) [reply Kernel/Devices/OffsetDiskDevice.h]: Then let's call it `DiskPartition` right away and over time we can massage it into being more of a fully-fledged partition object. :)

Comment (deoxxa) [reply Kernel/Devices/OffsetDiskDevice.h]: Done and done!

**Pull request merged**